### PR TITLE
docs: Bring basic docs structure from starbase

### PR DESCRIPTION
**Pull request by lengau**
I'm marking this as a draft PR since Diataxis documentation should always be complete and this is by definition not, but it's a starting point for writing the documentation.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = "starcraft"
+copyright = "2023, Canonical"
+author = "Canonical"
+
+# region General configuration
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration
+
+extensions = [
```
Comment (tigarmo) [inline docs/conf.py:16]: where are the python packages for these declared?

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
+.. starcraft documentation root file
```
Comment (tigarmo) [inline docs/index.rst:1]: should we rename all instances of `starcraft` here to `craft-archives`? Or is that going to happen in a future change?

**Review by tigarmo — APPROVED** [review]
thanks!

**Comment by tigarmo** [comment]
@lengau are you OK with un-drafting and merging this? Is there anything else you want to do here?

**Comment by lengau** [comment]
@tigarmo Done!

**Pull request merged**